### PR TITLE
Update coding exercise 5

### PR DESCRIPTION
**Pull request by maryx**
Update the 5th coding exercise.

### CODE REVIEW

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart [hunk 1]
```diff
@@ -56,68 +51,47 @@ class _CategoryRouteState extends State<CategoryRoute> {
     Colors.red,
   ];
 
-  /// Returns a list of mock [Unit]s
+  /// Returns a list of mock [Unit]s.
```
Comment (yjbanov) [inline course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: BTW, see also [List.generate](https://api.dartlang.org/stable/1.24.3/dart-core/List/List.generate.html).

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart [hunk 2]
```diff
@@ -56,68 +51,47 @@ class _CategoryRouteState extends State<CategoryRoute> {
...
-            new Expanded(child: _categories[i + 1]),
-          ],
-        ),
-      ));
-      if (i + 2 < _categories.length) {
-        rows.add(_bottomPadding);
-      }
-    }
-    return rows;
+  /// Makes the correct number of rows for the List View.
```
Comment (yjbanov) [inline course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: nit: no need to capitalize list view (it's not a proper noun).

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart [hunk 3]
```diff
@@ -56,68 +51,47 @@ class _CategoryRouteState extends State<CategoryRoute> {
...
+  ///
+  /// For portrait, we use a ListView.
```
Comment (yjbanov) [inline course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: Wrap "ListView" in square brackets so it's linkable and lintable.

File: course/05_stateful_widgets/task_05_stateful_widgets/lib/category_route.dart [hunk 4]
```diff
@@ -52,43 +47,26 @@ class CategoryRoute extends StatelessWidget {
...
-        ),
-      ));
-      if (i + 2 < categories.length) {
-        rows.add(_bottomPadding);
-      }
-    }
-    return rows;
+  /// Makes the correct number of rows for the List View.
+  ///
+  /// For portrait, we use a ListView.
```
Comment (yjbanov) [inline course/05_stateful_widgets/task_05_stateful_widgets/lib/category_route.dart]: ditto

**Comment by yjbanov** [comment]
lgtm

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart (hunk 2 above)
Comment (maryx) [reply course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: Ah, it's hard to catch all of these, thanks!

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart (hunk 3 above)
Comment (maryx) [reply course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: Thanks!

File: course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart (hunk 1 above)
Comment (maryx) [reply course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: Great, I'll make a PR for this.

Comment (maryx) [reply course/05_stateful_widgets/solution_05_stateful_widgets/lib/category_route.dart]: Actually it's small enough that I will just include it here in this PR. Thanks!

**Pull request merged**